src/attr.rs

```diff
@@ -116,7 +116,9 @@ fn format_derive(
                 |span| span.lo(),
                 |span| span.hi(),
                 |span| Some(context.snippet(*span).to_owned()),
-                attr.span.lo(),
+                // We update derive attribute spans to start after the opening '('
+                // This helps us focus parsing to just what's inside #[derive(...)]
+                attr.span.lo() + BytePos("#[derive(".len() as u32),
```

Comment:
Excellent, this is precisely what I had in mind! This type of offset derivation is so common that we actually have a lot of utilities that make this easy to do, and account for various edge cases and complexities (whitespace, comments, etc.) so we'll want to apply the same idea here, but utilize the helper instead. We can also just look for the opening paren in this case, so that we don't have to worry about anything weird but technically valid like `#[ derive    (` type scenaro

e.g. (disclaimer, typed inline so may need to be tweaked and/or have the SpanUtils imported from source_map)
```suggestion
                context.snippet_provider.span_after(attr.span, "("),
```